### PR TITLE
Add an example for clearing default configuration sources

### PR DESCRIPTION
**Pull request by UncleDave**
Addresses #15370

Added an example that shows how to clear the default configuration sources provided by `Host.CreateDefaultBuilder` to the _Default configuration_ section.

### CODE REVIEW

File: aspnetcore/fundamentals/configuration/index.md [hunk 1]
```diff
@@ -75,11 +75,13 @@ The following applies to apps using the [Generic Host](xref:fundamentals/host/ge
 To remove the providers added by `CreateDefaultBuilder`, call `Sources.Clear()` first:
 
 ```csharp
-.ConfigureAppConfiguration((hostingContext, config) =>
-{
-    config.Sources.Clear();
-    // Add providers here
-})
+Host.CreateDefaultBuilder(args)
```
Comment (UncleDave) [inline aspnetcore/fundamentals/configuration/index.md]: I considered that it's a bit context-less in this section :D Whereas the context was clearer down in the _ConfigureAppConfiguration_ section.

**Comment by UncleDave** [comment]
@guardrex Other than the difference of `Host` vs `WebHost` is this not just as relevant to _<3.0_?

**Comment by guardrex** [comment]
Yes, `ConfigureAppConfiguration` was a thing back in 2.1, so we're good there. I only did that as a prelude to perhaps having `WebHost.CreateDefaultBuilder` in the <3.0 piece in the *Default configuration* section, but I agree with u ... this is better the way that you had it and it's better down here. Sorry for the churn.

**Comment by UncleDave** [comment]
All good, I think we needed to move it up in order to realise it was better off down below with the rest of the _ConfigureAppConfiguration_ stuff.

**Comment by guardrex** [comment]
These docs are *complicated!* :smile: ..... I'm a small green dinosaur in a **BIG** framework here. :smile:

**Comment by guardrex** [comment]
The main thing in my head right now is the "flow" concept. I think that requires more work in a general way ... it has to be a little less *implied* and a little more *explicit* about the **_flow_** of configuration. I'll take a look at cross-links, too. I'll consider floating the manual host builder with a built config, but I also have that feel'in that engineering isn't going to like that too much. We'll see. I'll ping u on that PR when it goes up. Thanks again for the discussion and your help with the docs.

**Pull request merged**